app/templates/components/forms/orders/guest-order-form.hbs

```diff
@@ -15,7 +15,7 @@
       {{input type='password' name="password" value=password}}
     </div>
     <div class="ui center aligned segment basic">
-      <a href="{{href-to 'reset-password'}}" class="text muted" target="_blank" rel="noopener noreferrer"> {{t 'Forgot your password ?'}}</a>
+      <a href="{{href-to 'reset-password'}}" class="text muted" onclick="setTimeout(location.reload.bind(location), 1)"> {{t 'Forgot your password ?'}}</a>
```

Comment:
How does this solve the issue?

Reply:
on `omclick`, the url is changed by the href and then the page is reloaded.

Reply:
This is not the solution at all

Reply:
How come?

Reply:
The issue says to open the link in the same page and you write `onclick="setTimeout(location.reload.bind(location), 1)". I would really want to know what was the thought process on this conclusion. Every normal link on pages opens in the same window. Have you seen this written anywhere else?

Reply:
The earlier implementation, which was done like any other link on the page, opened the url on the same page but lead to the issue #2753 which was then solved by adding `target="_blank" rel="noopener noreferrer"`, pr #2827. 
But since this outcome was not desired and we had to do something that solves the issue as well, so what I had done is call a refresh action as soon as the link opens up in the same window so as to avoid the not responding screen flashing in the issue #2753

Reply:
The solution in #2827 was wrong. Fix it correctly

Reply:
Like how? What should be the required outcome? Should it open in the same window? If so, what should be my plan of action for it (if this is not the right way)?

Reply:
Remove target blank and fix the freezing issue

Reply:
Can I use `onclick="window.location.href='/reset-password'"` instead, then? I did not find any other alternative, and fix too.

Reply:
No, that shouldn't be needed. CSP is just a warning and if you pull latest changes, it'll be gone. Find out why does the window freeze.